api/openapi/specs/engine_service.yaml

```diff
@@ -0,0 +1,149 @@
+openapi: 3.0.0
+
+info:
+  title: Engine API
+  version: 0.1.0
+
+paths:
+  /engine/info:
+    get:
+      summary: Get Engine Info
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: './api_types.yaml#/components/schemas/EngineInfo'
+
+  /enclaves:
+    get:
+      summary: Get Enclaves
+      responses:
+        default:
+          $ref: '#/components/responses/NotOk'
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties:
+                  $ref: './api_types.yaml#/components/schemas/EnclaveInfo'
+                
+    post:
+      summary: Create Enclave
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: './api_types.yaml#/components/schemas/CreateEnclave'
+      responses:
+
+        default:
+          $ref: '#/components/responses/NotOk'
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: './api_types.yaml#/components/schemas/EnclaveInfo'
+                
+    delete:
+      summary: Delete Enclaves
+      parameters:
+        - $ref: './api_types.yaml#/components/parameters/remove_all'
```

Comment:
probably the description could describe what happens when the value is false, I think only stopped enclave will be removed, right?

Reply:
yes!